chutney/server/src/main/java/com/chutneytesting/security/infra/sso/OAuth2SsoSecurityConfiguration.java

```diff
@@ -0,0 +1,138 @@
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserRequest;
+import org.springframework.security.oauth2.client.userinfo.OAuth2UserService;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.www.BasicAuthenticationFilter;
+import org.springframework.security.web.servlet.util.matcher.MvcRequestMatcher;
+import org.springframework.web.servlet.handler.HandlerMappingIntrospector;
+
+@Configuration
+@Profile("sso-auth")
+@EnableWebSecurity
+@EnableMethodSecurity
+@EnableConfigurationProperties(OAuth2AuthorizationServerProperties.class)
+public class OAuth2SsoSecurityConfiguration {
+
+    @Value("${management.endpoints.web.base-path:/actuator}")
+    public static String ACTUATOR_BASE_URL;
+
+    @Value("${server.ssl.enabled:true}")
+    Boolean sslEnabled;
+
+    @Bean
+    public AuthenticationService authenticationService(Authorizations authorizations) {
+        return new AuthenticationService(authorizations);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public SsoOpenIdConnectConfig emptySsoOpenIdConnectConfig() {
+        return new SsoOpenIdConnectConfig();
+    }
+
+    @Bean
+    @Primary
+    SsoOpenIdConnectConfig ssoOpenIdConnectConfig(
+        @Value("${auth.sso.issuer}") String issuer,
+        @Value("${auth.sso.clientId}") String clientId,
+        @Value("${auth.sso.clientSecret}") String clientSecret,
+        @Value("${auth.sso.responseType}") String responseType,
+        @Value("${auth.sso.scope}") String scope,
+        @Value("${auth.sso.redirectBaseUrl}") String redirectUri,
+        @Value("${auth.sso.oidc}") Boolean oidc,
+        @Value("${auth.sso.ssoProviderName}") String ssoProviderName
+    ) {
+        return new SsoOpenIdConnectConfig(issuer,
+            clientId,
+            clientSecret,
+            responseType,
+            scope,
+            redirectUri,
+            ssoProviderName,
+            oidc);
+    }
+
+    @Bean
+    public OAuth2UserService<OAuth2UserRequest, OAuth2User> customOAuth2UserService(AuthenticationService authenticationService) {
+        return new OAuth2SsoUserService(authenticationService);
+    }
+
+    @Bean
+    public OAuth2TokenAuthenticationProvider tokenAuthenticationProvider(AuthenticationService authenticationService, ClientRegistrationRepository clientRegistrationRepository) {
+        return new OAuth2TokenAuthenticationProvider(customOAuth2UserService(authenticationService), clientRegistrationRepository.findByRegistrationId("my-provider"));
+    }
+
+    @Bean
+    public AuthenticationManager authenticationManager(OAuth2TokenAuthenticationProvider OAuth2TokenAuthenticationProvider) {
+        return new ProviderManager(Collections.singletonList(OAuth2TokenAuthenticationProvider));
+    }
+
+    @Bean
+    @Order(1)
```

Comment:
to avoid to handle two `SecurityFilterChain`, I prefer have always oAuth2TokenAuthentification and always send a token that can be generated by chutney server on external sso server.
This may need to change the existing auth mechanism.

Reply:
Humm you mean to always enable the sso ? And what would be the behaviour if there is no SSO configured ? This means making the SecurityFilterChain even more complex